### PR TITLE
feat: Implementing infinityModel at public session route

### LINKED ISSUE #4023: Month filter not working in session public page
Month filter is misbehaving in public session page

### PR DESCRIPTION
**Pull request by kushthedude**
Refers #3650
Fixes #4023
- Sessions were not rendering due to lack of space.
- Introduced infinity loader and checked the trigger offset, both are working quite well

### CODE REVIEW

File: app/routes/public/sessions/list.js [hunk 1]
```diff
@@ -16,169 +16,175 @@ export default Route.extend({
...
-              }
-            ]
-          }
-        ]
-      });
+              ]
+            }
+          ]
+        }
+      ];
```
Comment (iamareebjamal) [inline app/routes/public/sessions/list.js]: Massive duplication. All filters are almost same

File: app/templates/public/sessions/list.hbs [hunk 2]
```diff
@@ -10,4 +10,11 @@
   {{else}}
     <div class="ui disabled header">{{t 'No Sessions exist for this time period'}}</div>
   {{/each}}
+  {{#infinity-loader infinityModel=model.session triggerOffset=300 eventDebounce=50}}
+    <div class="center aligned five wide column">
+      <div class="ui loading very padded basic segment">
+      </div>
+    </div>
+    {{infintyModel.reachedInfinity}}
```
Comment (iamareebjamal) [inline app/templates/public/sessions/list.hbs]:
```suggestion
```

Comment (kushthedude) [reply app/templates/public/sessions/list.hbs]: Why ?

File: app/routes/public/sessions/list.js (hunk 1 above)
Comment (kushthedude) [reply app/routes/public/sessions/list.js]: They were implemented like this before, Also month was not working. 
Fixed them now

File: app/templates/public/sessions/list.hbs (hunk 2 above)
Comment (iamareebjamal) [reply app/templates/public/sessions/list.hbs]: Not needed

**Comment by iamareebjamal** [comment]
Goal is to improve the code as we move on. "It was implemented this way" doesn't really work

**Comment by kushthedude** [comment]
@iamareebjamal done. Please merge

File: app/routes/public/sessions/list.js [hunk 3]
```diff
@@ -16,169 +16,97 @@ export default Route.extend({
...
-              }
-            ]
+            name : 'starts-at',
+            op   : 'ge',
+            val  : moment().startOf('month').toISOString()
+          },
+          {
+            name : 'starts-at',
+            op   : 'lt',
+            val  : moment().endOf('month').toISOString()
```
Comment (iamareebjamal) [inline app/routes/public/sessions/list.js]: Still duplicated thrice

Comment (kushthedude) [reply app/routes/public/sessions/list.js]: How can I filter month, week, day better than this?

**Comment by kushthedude** [comment]
@iamareebjamal Please review now

File: app/routes/public/sessions/list.js [hunk 4]
```diff
@@ -16,169 +16,63 @@ export default Route.extend({
...
-            ]
-          }
-        ]
-      });
-    } else {
-      sessions = await this.store.query('session', {
-        filter: [
+            name : 'starts-at',
+            op   : 'ge',
+            val  : moment().startOf(`${params.session_status}`).toISOString()
```
Comment (iamareebjamal) [inline app/routes/public/sessions/list.js]: For 'day' it was 'today' I suppose

Comment (kushthedude) [reply app/routes/public/sessions/list.js]: There will be an extra conditional for today then.

Comment (iamareebjamal) [reply app/routes/public/sessions/list.js]:
```js
const period = params.session_status === 'today' ? 'day' : params.session_status;
```

Comment (kushthedude) [reply app/routes/public/sessions/list.js]: > ```js
> const period = params.session_status === 'today' ? 'day' : params.session_status;
> ```

Thanks, Done!

**Comment by kushthedude** [comment]
@iamareebjamal Please Review and Merge now!

File: app/routes/public/sessions/list.js [hunk 5]
```diff
@@ -16,169 +16,64 @@ export default Route.extend({
   },
   async model(params) {
     const eventDetails = this.modelFor('public');
-    let sessions =  null;
-    if (params.session_status === 'today') {
-      sessions = await this.store.query('session', {
-        filter: [
+    const period = params.session_status === 'today' ? 'day' : params.session_status;
+    let filterOptions = [
```
Comment (iamareebjamal) [inline app/routes/public/sessions/list.js]:
```suggestion
    const filterOptions = [
```

File: app/routes/public/sessions/list.js [hunk 6]
```diff
@@ -16,169 +16,64 @@ export default Route.extend({
   },
   async model(params) {
     const eventDetails = this.modelFor('public');
-    let sessions =  null;
-    if (params.session_status === 'today') {
-      sessions = await this.store.query('session', {
-        filter: [
+    const period = params.session_status === 'today' ? 'day' : params.session_status;
```
Comment (iamareebjamal) [inline app/routes/public/sessions/list.js]: Define it inside the if condition below

File: app/routes/public/sessions/list.js [hunk 7]
```diff
@@ -16,169 +16,64 @@ export default Route.extend({
...
-                    name : 'state',
-                    op   : 'eq',
-                    val  : 'accepted'
-                  }
-                ]
-              }
-            ]
+            name : 'starts-at',
+            op   : 'lt',
+            val  : moment().endOf(`${period}`).toISOString()
```
Comment (iamareebjamal) [inline app/routes/public/sessions/list.js]:
```suggestion
            val  : moment().endOf(period).toISOString()
```

File: app/routes/public/sessions/list.js (hunk 5 above)
Comment (kushthedude) [reply app/routes/public/sessions/list.js]: Filter options are changing in the if conditional, const will not be useful then.

File: app/routes/public/sessions/list.js [hunk 8]
```diff
@@ -16,169 +16,64 @@ export default Route.extend({
...
-            ]
-          }
-        ]
-      });
-    } else {
-      sessions = await this.store.query('session', {
-        filter: [
+            name : 'starts-at',
+            op   : 'ge',
+            val  : moment().startOf(`${period}`).toISOString()
```
Comment (iamareebjamal) [inline app/routes/public/sessions/list.js]:
```suggestion
            val  : moment().startOf(period).toISOString()
```

File: app/routes/public/sessions/list.js (hunk 5 above)
Comment (iamareebjamal) [reply app/routes/public/sessions/list.js]: Try it

Comment (kushthedude) [reply app/routes/public/sessions/list.js]: ESLint passed 😮

**Comment by kushthedude** [comment]
@iamareebjamal Check now

**Pull request merged**